sources/packages/backend/workflow/test/2023-2024/fulltime-assessment/eligibility/fulltime-assessment-2023-2024-eligibility-CSGD.e2e-spec.ts

```diff
@@ -0,0 +1,104 @@
+        DependentEligibility.Eligible0To18YearsOld,
+      ),
+      createFakeStudentDependentEligible(
+        DependentEligibility.Eligible18To22YearsOldAttendingHighSchool,
+      ),
+      createFakeStudentDependentEligible(
+        DependentEligibility.Eligible18To22YearsOldDeclaredOnTaxes,
+      ),
+      createFakeStudentDependentEligible(
+        DependentEligibility.EligibleOver22YearsOld,
+      ),
+      createFakeStudentDependentNotEligible(
+        DependentEligibility.Eligible0To18YearsOld,
+      ),
+      createFakeStudentDependentNotEligible(
+        DependentEligibility.Eligible18To22YearsOldAttendingHighSchool,
+      ),
+      createFakeStudentDependentNotEligible(
+        DependentEligibility.Eligible18To22YearsOldDeclaredOnTaxes,
+      ),
+      createFakeStudentDependentNotEligible(
+        DependentEligibility.EligibleOver22YearsOld,
+      ),
+    ];
+    // Act
+    const calculatedAssessment = await executeFulltimeAssessmentForProgramYear(
+      PROGRAM_YEAR,
+      assessmentConsolidatedData,
+    );
+    // Assert
+    // calculatedDataTotalFamilyIncome must be below the threshold for
+    // dmnFullTimeAwardFamilySizeVariables.limitAwardCSGDThresholdIncome.
+    expect(calculatedAssessment.variables.calculatedDataTotalFamilyIncome).toBe(
+      assessmentConsolidatedData.studentDataTaxReturnIncome,
+    );
+    expect(calculatedAssessment.variables.calculatedDataFamilySize).toBe(5);
+    expect(calculatedAssessment.variables.awardEligibilityCSGD).toBe(true);
+    expect(
+      calculatedAssessment.variables.federalAwardNetCSGDAmount,
+    ).toBeGreaterThan(0);
+    expect(
+      calculatedAssessment.variables.provincialAwardNetCSGDAmount,
+    ).toBeGreaterThan(0);
+  });
+
+  it("Should determine CSGD as not eligible when financial need is at least $1 and total family income is above the threshold and eligible dependents are greater than 1.", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedFulltimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.studentDataTaxReturnIncome = 100000;
+    // Creates one eligible dependent (between 0-18 years old condition).
+    assessmentConsolidatedData.studentDataDependants = [
+      createFakeStudentDependentEligible(
+        DependentEligibility.Eligible0To18YearsOld,
+      ),
+    ];
+    // Act
+    const calculatedAssessment = await executeFulltimeAssessmentForProgramYear(
+      PROGRAM_YEAR,
+      assessmentConsolidatedData,
+    );
+    // Assert
+    expect(calculatedAssessment.variables.awardEligibilityCSGD).toBe(false);
+  });
+
+  it("Should determine CSGD as not eligible when financial need is at least $1 and total family income is above the threshold and eligible dependents is 0.", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedFulltimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.studentDataTaxReturnIncome = 32999;
```

Comment:
as per line 17, I think `32999` is under the threshold, but line 86 description does not match

Reply:
Thresholds are different accordingly to the family size.